thread.c

```diff
@@ -695,6 +700,17 @@ thread_do_start(rb_thread_t *th)
 }
 
 void rb_ec_clear_current_thread_trace_func(const rb_execution_context_t *ec);
+rb_control_frame_t *
```

Comment:
move this decl to vm_core.h.

Reply:
Good idea.

Reply:
I added

```ruby
void
rb_ec_initialize_vm_stack(rb_execution_context_t *ec, VALUE *stack, size_t size)
{
  rb_ec_set_vm_stack(ec, vm_stack, size);
  th->ec->cfp = (void *)(ec->vm_stack + ec->vm_stack_size);

  rb_vm_push_frame(ec,
      NULL, /* dummy iseq */,
      VM_FRAME_MAGIC_DUMMY | VM_ENV_FLAG_LOCAL | VM_FRAME_FLAG_FINISH | VM_FRAME_FLAG_CFRAME /* dummy frame */,
      Qnil /* dummy self */, VM_BLOCK_HANDLER_NONE /* dummy block ptr */,
      0 /* dummy cref/me */,
      0 /* dummy pc */, ec->vm_stack, 0, 0
  );
}
```

in `vm.c` and exposed it in `vm_core.h` then used it everywhere.